overlord/hookstate/hookmgr.go

```diff
@@ -256,24 +273,30 @@ func (m *HookManager) doRunHook(task *state.Task, tomb *tomb.Tomb) error {
 		return err
 	}
 
-	if hookExists {
-		output, err := runHook(context, tomb)
-		if err != nil {
-			if hooksup.TrackError {
-				trackHookError(context, output, err)
-			}
-			err = osutil.OutputErr(output, err)
-			if hooksup.IgnoreError {
-				task.State().Lock()
-				task.Errorf("ignoring failure in hook %q: %v", hooksup.Hook, err)
-				task.State().Unlock()
-			} else {
-				if handlerErr := context.Handler().Error(err); handlerErr != nil {
-					return handlerErr
-				}
-
-				return fmt.Errorf("run hook %q: %v", hooksup.Hook, err)
+	// some hooks get hijacked, e.g. the core configuration
+	var output []byte
+	if f := m.hijacked(hooksup.Hook, hooksup.Snap); f != nil {
```

Comment:
Given there is likely a change necessary anyway to address the error checking, I'll nitpick just slightly. :)

_"hijacked"_ feels like something that already took place, but really the whole workflow is exactly the same until we get right here in this line above. Considering that, it feels like it would be just slightly more clear if we used the "hijack" term in the present, as a request of something we'd like to happen, and use "hijacked", in the past, only once and if we're talking about something that did happen before. So _RegisterHijack,_ and perhaps _mustHijack_ instead of _isHijacked_, etc.

Again, this is getting deeply into nitpick territory, so feel free to ignore if you don't feel like it's worth it.

Reply:
I think when I suggested RegisterHijack it was indeed the name or present, seems I didn't notice the change to Hijacked when I re-reviewed. Anyway also +1 on switching from me.

Reply:
Indeed, thank you. This makes it clearer/cleaner!